src/validation.cpp

```diff
@@ -2701,22 +2702,22 @@ bool CheckBlock(const CBlock& block, CValidationState& state, bool fCheckPOW, bo
                 nHeight = (*mi).second->nHeight + 1;
         }
 
+        // Last output of Cold-Stake is not abused
+        if (IsPoS && !CheckColdStakeFreeOutput(block.vtx[1], nHeight)) {
+            mapRejectedBlocks.emplace(block.GetHash(), GetTime());
+            return state.DoS(0, false, REJECT_INVALID, "bad-p2cs-outs", false, "invalid cold-stake output");
+        }
+
+        // set Cold Staking Spork
+        fColdStakingActive = !sporkManager.IsSporkActive(SPORK_19_COLDSTAKING_MAINTENANCE);
```

Comment:
This must remain true during IBD (as per comment at line 2688), since we don't have historical records for sporks.